src/rust/src/encoder/g608.rs

```diff
@@ -0,0 +1,308 @@
+    }
+
+    Ok(())
+}
+
+/// Get line characters encoded according to context encoding
+pub fn get_line_encoded(
+    ctx: &encoder_ctx,
+    buffer: &mut [c_uchar],
+    line_num: usize,
+    data: &eia608_screen,
+) -> c_uint {
+    let mut buffer_pos = 0;
+    let line = &data.characters[line_num];
+
+    for i in line.iter().take(33) {
+        let bytes_written = match ctx.encoding {
+            ccx_encoding_type_CCX_ENC_UTF_8 => {
+                let utf8_packed = line21_to_utf8(*i);
+                let (byte_count, bytes) = if utf8_packed <= 0xff {
+                    (1, [(utf8_packed & 0xFF) as c_uchar, 0, 0])
+                } else if utf8_packed <= 0xffff {
+                    (
+                        2,
+                        [
+                            ((utf8_packed >> 8) & 0xFF) as c_uchar,
+                            (utf8_packed & 0xFF) as c_uchar,
+                            0,
+                        ],
+                    )
+                } else {
+                    (
+                        3,
+                        [
+                            ((utf8_packed >> 16) & 0xFF) as c_uchar,
+                            ((utf8_packed >> 8) & 0xFF) as c_uchar,
+                            (utf8_packed & 0xFF) as c_uchar,
+                        ],
+                    )
+                };
+
+                if buffer_pos + byte_count - 1 < buffer.len() {
+                    buffer[buffer_pos..(byte_count + buffer_pos)]
+                        .copy_from_slice(&bytes[..byte_count]);
+                    byte_count
+                } else {
+                    0
+                }
+            }
+            ccx_encoding_type_CCX_ENC_LATIN_1 => {
+                if buffer_pos < buffer.len() {
+                    let latin1_char = line21_to_latin1(*i);
+                    buffer[buffer_pos] = latin1_char as c_uchar;
+                    1
+                } else {
+                    0
+                }
+            }
+            ccx_encoding_type_CCX_ENC_UNICODE => {
+                if buffer_pos + 1 < buffer.len() {
+                    let ucs2_char = line21_to_ucs2(*i);
+                    // UCS-2 is 2 bytes, little-endian
+                    let bytes = ucs2_char.to_le_bytes();
+                    buffer[buffer_pos] = bytes[0] as c_uchar;
+                    buffer[buffer_pos + 1] = bytes[1] as c_uchar;
+                    2
+                } else {
+                    0
+                }
+            }
+            ccx_encoding_type_CCX_ENC_ASCII => {
+                if buffer_pos < buffer.len() {
+                    buffer[buffer_pos] = *i;
+                    1
+                } else {
+                    0
+                }
+            }
+            _ => {
+                0 // This should never be reached
+            }
+        };
+
+        buffer_pos += bytes_written;
+
+        // Break if we've run out of buffer space
+        if bytes_written == 0 {
```

Comment:
If you really want this case, move it above line 111. Also should we really just break out of for loop without any error? Asking because C code doesn't cover this case.